proposals/p1964.md

```diff
@@ -0,0 +1,234 @@
+Meaning the type depends on the the contents, so that `'c'` and `'b'`. However
+any `'\n'` and `'\u{A}'` would be of the same type, as they are the same unicode
+entities `%0A`. It is worth noting that `'\x0A'` will be slightly different as
+any character following the `\x...` sequence representation are not the same as
+a Unicode character. For example:
+
+```
+var c1: EBCDICChar = '\x0A';
+var c2: EBCDICChar = '\n';
+```
+
+Here we assume that `c1` will be the EBCDIC RPT character (0xA) and for `c2` to
+be the EBCDIC NL character (0x15)
+
+### Operations
+
+From the previous examples, we should be able to use the following operators:
+
+-   Comparison: `<`, `>`, `<=`, `>=` `==`
+-   Plus: `+`. However the behaviour of this operator is distinctly compared to
+    its use with Strings. Rather then a concatenation, this will add the value
+    of the two characters:
+    -   If the `+` is used between a character literal and an integer literal,
+        this should produce a character literal if the result fits into a
+        Unicode code point or for a unicode hex code like `'\xAB'`, a single
+        byte.
+    -   If the `+` is used between a character literal and an integer
+        non-literal this will produce and error.
+    -   If the `+` is used between two character literals this will produce and
+        error.
+-   Subtract: `-` Similar to the plus operator, this will subtract the value of
+    the two characters.
+
+    -   If the `-` is used between a character literal and an integer literal,
+        this should produce a character literal.
+    -   If the `-` is used between two character literals this should produce
+        and integer literal.
+    -   If the `-` is used between a character literal and an integer
+        non-literal this will produce and error.
+
+    In other use cases where the `+` and `-` operators are used, in theory we
+    should convert the character literal to some other type first. For example
+    if we want to call `x - 'a'`, where `w` is of type `char` we would want to
+    convert the `'a'` literal to a `char`.
+
+### Encoding
+
+Character literals written will be UTF-8 encoded, as all of Carbon source code
+is UTF-8 encoded. See
+[Unicode source files](https://github.com/carbon-language/carbon-lang/blob/trunk/proposals/p0142.md#character-encoding)
+
+## Rationale
+
+This proposal supports the goal of making Carbon code
+[easy to read, understand, and write](/docs/project/goals.md#code-that-is-easy-to-read-understand-and-write).
+Adding support for a specific character literal supports clean, readable,
+concises use and is a much more familiar concept that will make it easier to
+adopt Carbon coming from other languages. For Example when working with literals
+represented as hex, having a distinct character literal type makes this much
+simpler to work:
+
+```
+fn getSingleEAcuteHex(var combinedAcute: EBCDICChar) -> EBCDICChar {
```

Comment:
Is this a valid example to show the advantages of characters vs strings? Any suggestions on a better approach?

Reply:
Maybe use something like the example from the "Problem" section? I think the use of `EBCDICChar` here isn't really helping to motivate this choice. Pointing out that we can give operators different meanings for character literals versus strings seems useful, eg `'a' + 1 == 'b'` but `"a" + 1` should be a type error.

Reply:
That's a good point essentially not supporting `x/...` and enforcing hex code points to be written with the `0xAB` notation. off the top of my head I don't know of a good example/advantage here. @zygoloid, what are your thoughts on this?

Reply:
I think this was meant to be a reply to @josh11b's comment on line 275; replying there.